### PR TITLE
Make `useSearchField` behave close to browser

### PR DESCRIPTION
**Pull request by QzCurious**
This commit changes how `useSearchField` behave. It previously rely on `onSubmit` prop to get "Enter submission". For now, "Enter" would trigger a normal form submission by default browser behavior. For backward compatibility, if onSubmit prop is provide, `onSubmit` is called and default form submission is prevented.

related #6332

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/WAI/ARIA/apg/)

## 📝 Test Instructions:

Just render a `SearchField` and test it with/without `onSubmit`:

_(code modified form packages/react-aria-components/stories/SearchField.stories.tsx)_

```tsx
import {Button, Input, Label, SearchField} from 'react-aria-components';
import {classNames} from '@react-spectrum/utils';
import React from 'react';
import styles from '../example/index.css';

export default {
  title: 'React Aria Components'
};

export const SearchFieldExample = () => {
  return (
    <form
      onSubmit={(e) => {
        alert('browser default form submit');
        e.preventDefault();
      }}>
      <SearchField
        // onSubmit={() => alert('onSubmit prop')}
        className={classNames(styles, 'searchFieldExample')}
        data-testid="search-field-example">
        <Label>Search</Label>
        <Input />
        <Button>✕</Button>
      </SearchField>
    </form>
  );
};
```

https://github.com/adobe/react-spectrum/assets/38932402/b61941d7-9ff4-42fb-aecd-1515d32863d3

## 🧢 Your Project:

https://github.com/QzCurious/react-spectrum/tree/fix/useSearchField

### CODE REVIEW

File: packages/@react-aria/searchfield/test/useSearchField.test.js [hunk 1]
```diff
@@ -67,12 +67,10 @@ describe('useSearchField hook', () => {
         onSubmit.mockClear();
       });
 
-      it('preventDefault is called for Enter and not Escape', () => {
+      it('preventDefault is not called for Escape', () => {
```
Comment (ktabors) [inline packages/@react-aria/searchfield/test/useSearchField.test.js:70]: Can we combine this test and the following?
```suggestion
      it('preventDefault and stopPropagation are not called for Escape', () => {
```

**Review by ktabors — COMMENTED** [review]
Thanks for updating the tests.

**Review by snowystinger — DISMISSED** [review]

File: packages/@react-aria/searchfield/test/useSearchField.test.js (hunk 1 above)
Comment (QzCurious) [reply packages/@react-aria/searchfield/test/useSearchField.test.js]: I've merged the two test cases.

**Review by snowystinger — APPROVED** [review]

**Review by LFDanLu — APPROVED** [review]
Verified that the backcompat works and that omitting onSubmit properly causes native form validation, thanks for the fix!

**Pull request merged**